deploy/docker/templates/supervisord/application_process/editor.conf

```
@@ -5,8 +5,8 @@ autostart=true
 autorestart=true
 startsecs=0
 startretries=3
-stdout_logfile=/appsmith-stacks/logs/%(program_name)s/access.log
-stderr_logfile=/appsmith-stacks/logs/%(program_name)s/error.log
+stdout_logfile=/appsmith-stacks/logs/%(program_name)s/access%(ENV_HOSTNAME)s.log
```

```suggestion
stdout_logfile=/appsmith-stacks/logs/%(program_name)s/access-%(ENV_HOSTNAME)s.log
```

Minor: Adding a hyphen to differentiate between the service & hostname in the file. Please make this change in all the log configurations.

Also, I'm curious. What does `s` at the end of the string specify? Is it to convert the hostname to a string?

---

The problem with adding `-` is that when we try deploying appsmith on instances where env variable HOSTNAME is not present the file_name would be 
`backend-.log` which doesnt look good i feel. So, removed the -. if you suggest to keep - we shall 

`s` i am guessing its like format-specifiers only

---

It's Python's standard extension to `printf`-style formatting. Like `"hello %(name)s" % {"name": "Shri"}` is the same as `"hello %s" % "Shri"`.

---

Is it possible add a `-` to the `HOSTNAME` env variable itself, if it isn't empty? Sorry I thought that was how this was implemented. The log file names look quite amateurish without a dash. :|

---

Just realized, I _think_ `HOSTNAME` is a standard env variable. Let's not change it then.

Perhaps, something like:

```sh
APPSMITH_LOG_SUFFIX=""
if [[ -n "${HOSTNAME-}" ]]; then
  APPSMITH_LOG_SUFFIX="-$HOSTNAME"
fi
```

And then use `APPSMITH_LOG_SUFFIX` instead?

Also, if `HOSTNAME` is a standard env variable, it will _always_ be present, right? Like, the Docker container is always going to be an Ubuntu system, and whether it runs on Kubernetes or not, if Ubuntu sets `HOSTNAME`, it'll always be available no? In that case, should we just go with Arpit's suggested change in the first comment in this thread?